src/ai/storage.py

```diff
@@ -138,20 +136,39 @@ async def store_drone(message: discord.Message, message_copy=None):
         await message.channel.send(f'Drone with ID {target_id} could not be found.')
         return False
 
+    if is_free_storage(drone_to_store):
+        await initiate_drone_storage(drone_to_store, drone_id, target_id, time, purpose, message)
+        return False
+    else:
+        # check if initiator is allowed to store drone
+        trusted_users = get_trusted_users(drone_to_store.id)
+        initiator = fetch_drone_with_drone_id(drone_id)
+
+        # proceed if allowed, send error message if not
+        if (initiator.id == (HIVE_MXTRESS_USER_ID or drone_to_store.id)) or (initiator.id in trusted_users):
+            await initiate_drone_storage(drone_to_store, drone_id, target_id, time, purpose, message)
+        else:
+            await message.channel.send(f"Drone {target_id} can only be stored by its trusted users or the Hive Mxtress. It has not been stored.")
+            return False
```

Comment:
This return is not necessary as the program flow will continue to another `return False`  in line 153 without it as well.